shaptools/saputils.py

```diff
@@ -0,0 +1,67 @@
+"""
+Module to utilize SAP Technology components
+
+:author: sisingh
+:organization: SUSE LLC
+:contact: sisingh@suse.com
+
+:since: 2020-03-12
+"""
+#TODO: Add support for other SAPCAR functionalties apart from extraction
+
+from __future__ import print_function
+
+import os
+
+from shaptools import shell
+
+
+class SapUtilsError(Exception):
+    """
+    Error during SapUtils command execution
+    """
+
+
+class FileDoesNotExistError(Exception):
+    """
+    Error when the specified files does not exist
+    """
+
+
+def extract_sapcar_file(sapcar_exe, sar_file, **kwargs):
+    """
+    Execute SAPCAR command to decompress a SAP CAR or SAR archive files.
+    If user and password are provided it will be executed with this user.
+
+    Args:
+        sapcar_exe(str): Path to the SAPCAR executable
+        sar_file (str): Path to the sar file to be extracted
+        options (str, opt): Additional options to SAPCAR command
+        extract_dir (str, opt): Directory where archive will be extracted. It creates the dir
+            if the path doesn't exist. If it's not set the current dir is used
+        user (str, opt): User to execute the SAPCAR command
+        password (str, opt): User password
+        remote_host (str, opt): Remote host where the command will be executed
+    """
+    if not os.path.isfile(sapcar_exe):
+        raise FileDoesNotExistError('SAPCAR executable \'{}\' does not exist'.format({sapcar_exe}))
+    if not os.path.isfile(sar_file):
+        raise FileDoesNotExistError('The SAR file \'{}\' does not exist'.format(sar_file))
+
+    options = kwargs.get('options', None)
+    extract_dir = kwargs.get('extract_dir', None)
+    user = kwargs.get('user', None)
+    password = kwargs.get('password', None)
+    remote_host = kwargs.get('remote_host', None)
+
+    extract_dir_str = '-R {}'.format(extract_dir) if extract_dir else ''
+    options_str = '{}'.format(options) if options else ''
+
+    cmd = '{sapcar_exe} -xvf {sar_file} {options_str} {extract_dir_str}'.format(
```

Comment:
If you add the commented spaces in the previous lines, we could change this to:
`{sapcar_exe} -xvf {sar_file}{options_str}{extract_dir_str}'`
This way we wouldn't have strange space characters hanging there

Reply:
Yes I guess I can do it this way